src/tlskdf.rs

```diff
@@ -200,6 +205,60 @@ impl TLSPRFMechanism {
     }
 }
 
+#[cfg(feature = "fips")]
+struct FIPSSelftest {
+    result: CK_RV,
+}
+
+#[cfg(feature = "fips")]
+static TLS_PRF_SELFTEST: Lazy<FIPSSelftest> = Lazy::new(|| {
+    let mut status = FIPSSelftest {
+        result: CKR_FIPS_SELF_TEST_FAILED,
+    };
+
+    /* Test vector taken from OpenSSL selftest */
+    let prf: CK_MECHANISM_TYPE = CKM_SHA256_HMAC;
+    let secret = hex::decode("202c88c00f84a17a20027079604787461176455539e705be730890602c289a5001e34eeb3a043e5d52a65e66125188bf").unwrap();
+    let label: &[u8] = b"key expansion";
+    let randoms = hex::decode("ae6c806f8ad4d80784549dff28a4b58fd837681a51d928c3e30ee5ff14f3986862e1fd91f23f558a605f28478c58cf72637b89784d959df7e946d3f07bd1b616").unwrap();
+    let mut seed = Vec::<u8>::with_capacity(label.len() + randoms.len());
+    seed.extend_from_slice(&label);
+    seed.extend_from_slice(&randoms);
+
+    let expect = hex::decode("d06139889fffac1e3a71865f504aa5d0d2a2e89506c6f2279b670c3e1b74f531016a2530c51a3a0f7e1d6590d0f0566b2f387f8d11fd4f731cdd572d2eae927f6f2f81410b25e6960be68985add6c38445ad9f8c64bf8068bf9a6679485d966f1ad6f68b43495b10a683755ea2b858d70ccac7ec8b053c6bd41ca299d4e51928").unwrap();
+
+    /* mock key */
+    let mut key = Object::new();
+    key.set_attr(Attribute::from_ulong(CKA_CLASS, CKO_SECRET_KEY))
+        .unwrap();
+    key.set_attr(Attribute::from_ulong(CKA_KEY_TYPE, CKK_GENERIC_SECRET))
+        .unwrap();
+    key.set_attr(Attribute::from_bytes(CKA_VALUE, secret.clone()))
+        .unwrap();
+    key.set_attr(Attribute::from_ulong(
+        CKA_VALUE_LEN,
+        secret.len() as CK_ULONG,
+    ))
+    .unwrap();
+    key.set_attr(Attribute::from_bool(CKA_DERIVE, true))
+        .unwrap();
+
+    let mech = test_get_hmac(prf);
+
+    let mut tlsprf = match TLSPRF::init(&key, &mech, prf) {
+        Ok(a) => a,
+        Err(_) => return status,
+    };
+    let out = match tlsprf.finish(&seed, expect.len()) {
```

Comment:
Is there any particular reason why TLSPRF::finish takes label and seed as one Vec instead of two parameters as defined in RFC 5246?